### DISCUSSION TITLE
Updated example needed for Uno WebAssembly App using Azure Web Apps

### DISCUSSION THREAD

**Discussion by hbraasch** [unoplatform/uno#18911] [Q&A] [answered]
Can somebody please provide an updated example on how to change the Github Action yml file, when hosting a WebAssembly app on Azure Static Web Apps. The reason I have problems are:

All the examples I've found are for the old Uno solution file structure, assuming the Uno generated WebAssembly App gets created under the folder [MyApp/MyApp.Wasm] directory (with its own [.proj] file. Nowadays Uno only generates a single [.proj] file and it needs a different build instruction (need to change to "publish") in order to generate the correct output files. 

I've got it to work, but with a hack:

I've generated an Uno application named UnoApp2 using Visual Studio.
Hack: I had to remove all frameworks from the .proj file and only leave net8.0-browserwasm, otherwise the Github build was missing workloads...

![image](https://github.com/user-attachments/assets/88fe95ff-5e87-4280-90bc-bbdda1ff3b98)

This is what then remaines in the line :
`<TargetFrameworks>net8.0-browserwasm</TargetFrameworks>`
 I then pushed the solution to Github while inside Visual Studio. The [.proj] file then resides underneath the root folder of the Github repo in [UnoApp2/UnoApp2]
I've done all the prep in Azure to set up the Static Web Page with Github as repository. This appears to work fine.
I've then edited the Github actions file by adding:

      - name: Setup dotnet
        uses: actions/setup-dotnet@v1.7.2
        with:
         dotnet-version: '8.0'
      - run: |
          cd UnoApp2
          dotnet publish -f net8.0-browserwasm -c Release

and changed the app location to:

`app_location: "UnoApp2/bin/Release/net8.0-browserwasm/publish/wwwroot" # App source code path`

As I say, this works, but I need to remove the unnecessary frameworks.

Is there a way to have `dotnet build -f net8.0-browserwasm -c Release` not require me to delete all the other frameworks?

Or is there a complete beter way overall e.g. the dotnet build => dotnet publish change is unnecessary?

Help shall be much appreciated

Thanks

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report. You can add the following parameter:

```
-p:TargetFrameworks=net8.0-browserwasm
```

to the command line to avoid requiring additional targets to be restored.

**Comment by hbraasch** [comment]
Thanks. It works a charm.
Much appreciated
